backend/b2/b2.go

```diff
@@ -548,12 +548,12 @@ func (f *Fs) newObjectWithInfo(ctx context.Context, remote string, info *api.Fil
 	if info != nil {
 		err := o.decodeMetaData(info)
 		if err != nil {
-			return nil, err
+			return o, err
```

Comment:
I think this change and the one below will break the integration tests which expect a `nil` object if `NewObject` fails.